### PR TITLE
Media Library: Fix usage of `aria-label` and `alt` attributes on videos/images

### LINKED ISSUE #7574: Media Library: Incorrect use of `aria-label`
## Bug Description

Images and videos in the media library incorrectly have `aria-label` attributes.

Both should only have `alt` attributes.

Offending code:

https://github.com/google/web-stories-wp/blob/4c891bd2b6b10815c72b06d7bbccd15718da79b1/assets/src/edit-story/components/library/panes/media/common/innerElement.js#L135-L167

## Expected Behaviour

No usage of `aria-label`. Just `alt`

## Screenshots

Screenshot 2021-05-18 at 23 14 35

Screenshot 2021-05-18 at 23 14 15

## Additional Context

- Plugin Version: 1.7
- WordPress Version: 5.7
- Operating System: macOS
- Browser: Chrome

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Implementation Brief

* Remove the two `aria-label` props in the code section shared above

### PR DESCRIPTION
**Pull request by Sreelayavuyyuru**
## Summary

This PR solves the issue #7574. I have removed the aria-label to replace it with alt. 

### QA

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [x] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #7574

### CODE REVIEW

**Comment by Sreelayavuyyuru** [comment]
> Thanks for your pull request. It looks like this may be your first contribution to a Google open source project (if not, look below for help). Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> 📝 **Please visit https://cla.developers.google.com/ to sign.**
> 
> Once you've signed (or fixed any issues), please reply here with `@googlebot I signed it!` and we'll verify it.
> 
> #### What to do if you already signed the CLA
> ##### Individual signers
> * It's possible we don't have your GitHub username or you're using a different email address on your commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> 
> ##### Corporate signers
> * Your company has a Point of Contact who decides which employees are authorized to participate. Ask your POC to be added to the group of authorized contributors. If you don't know who your Point of Contact is, direct the Google project maintainer to [go/cla#troubleshoot](http://go/cla#troubleshoot) ([Public version](https://opensource.google/docs/cla/#troubleshoot)).
> * The email used to register you as an authorized contributor must be the email used for the Git commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> * The email used to register you as an authorized contributor must also be [attached to your GitHub account](https://github.com/settings/emails).
> 
> ℹ️ **Googlers: [Go here](https://goto.google.com/prinfo/https%3A%2F%2Fgithub.com%2Fgoogle%2Fweb-stories-wp%2Fpull%2F7923) for more info**.

@googlebot I signed it!

File: assets/src/edit-story/components/library/panes/media/common/innerElement.js [hunk 1]
```diff
@@ -154,11 +154,11 @@ function InnerElement({
   const imageProps = {
     ...commonImageProps,
     src: thumbnailURL,
-    'aria-label': alt,
+    alt: alt,
```
Comment (spacedmonkey) [inline assets/src/edit-story/components/library/panes/media/common/innerElement.js]:
```suggestion
    alt,
```

File: assets/src/edit-story/components/library/panes/media/common/innerElement.js [hunk 2]
```diff
@@ -154,11 +154,11 @@ function InnerElement({
...
   };
   const videoProps = {
     ...commonProps,
-    'aria-label': alt,
+    alt: alt,
```
Comment (spacedmonkey) [inline assets/src/edit-story/components/library/panes/media/common/innerElement.js]:
```suggestion
```

Videos dont need alts

Comment (Sreelayavuyyuru) [reply assets/src/edit-story/components/library/panes/media/common/innerElement.js]: Okay, got it!

File: assets/src/edit-story/components/library/panes/media/common/innerElement.js [hunk 3]
```diff
@@ -154,11 +154,10 @@ function InnerElement({
   const imageProps = {
     ...commonImageProps,
     src: thumbnailURL,
-    'aria-label': alt,
+    alt,
```
Comment (swissspidy) [inline assets/src/edit-story/components/library/panes/media/common/innerElement.js]: `alt` is already included here as part of `commonImageProps` -> `commonProps`, so this is redundant here.

File: assets/src/edit-story/components/library/panes/media/common/innerElement.js [hunk 4]
```diff
@@ -154,11 +154,10 @@ function InnerElement({
...
   };
   const videoProps = {
     ...commonProps,
-    'aria-label': alt,
```
Comment (swissspidy) [inline assets/src/edit-story/components/library/panes/media/common/innerElement.js:162]: I think we want to keep this here actually, but don't want `alt`.

So

```
'aria-label': alt,
alt: null,
```

Comment (Sreelayavuyyuru) [reply assets/src/edit-story/components/library/panes/media/common/innerElement.js:162]: Okay, I have changed it. Does that look okay?

File: assets/src/edit-story/components/library/panes/media/common/innerElement.js (hunk 3 above)
Comment (Sreelayavuyyuru) [reply assets/src/edit-story/components/library/panes/media/common/innerElement.js]: I have now removed the redundant alt here.

**Review by swissspidy — APPROVED** [review]
Thanks a lot!

**Comment by swissspidy** [comment]
Looks like some tests are failing now.

The 4 tests in https://github.com/google/web-stories-wp/blob/a2ad2e45336f5a20ff87b68425b2e7fc181323dd/assets/src/edit-story/components/library/test/mediaElement.js need to be updated to use `getByAltText` instead of `getByAriaLabel`. This should make the tests pass.

1. Add `import { render, screen } from '@testing-library/react';`
2. Replace any `getByAriaLabel` usage with `screen.getByAltText`
3. Don't destructure `getByAriaLabel` from `renderMediaElement`

So for example the first test would look a bit like this:

```js
renderMediaElement(resource, 'local');
expect(screen.queryByAltText('More')).not.toBeInTheDocument();

const element = screen.getByAltText('image :)');
Simulate.focus(element);

expect(screen.getByAltText('More')).toBeInTheDocument();
```

The following test in resourceLoading.karma.js also needs to be updated to use `fixture.screen.getByAltText()`:

https://github.com/google/web-stories-wp/blob/a2ad2e45336f5a20ff87b68425b2e7fc181323dd/assets/src/edit-story/elements/image/karma/resourceLoading.karma.js#L43-L45

**Comment by Sreelayavuyyuru** [comment]
> Looks like some tests are failing now.
> 
> The 4 tests in https://github.com/google/web-stories-wp/blob/a2ad2e45336f5a20ff87b68425b2e7fc181323dd/assets/src/edit-story/components/library/test/mediaElement.js need to be updated to use `getByAltText` instead of `getByAriaLabel`. This should make the tests pass.
> 
> 1. Add `import { render, screen } from '@testing-library/react';`
> 2. Replace any `getByAriaLabel` usage with `screen.getByAltText`
> 3. Don't destructure `getByAriaLabel` from `renderMediaElement`
> 
> So for example the first test would look a bit like this:
> 
> ```js
> renderMediaElement(resource, 'local');
> expect(screen.queryByAltText('More')).not.toBeInTheDocument();
> 
> const element = screen.getByAltText('image :)');
> Simulate.focus(element);
> 
> expect(screen.getByAltText('More')).toBeInTheDocument();
> ```
> 
> The following test in resourceLoading.karma.js also needs to be updated to use `fixture.screen.getByAltText()`:
> 
> https://github.com/google/web-stories-wp/blob/a2ad2e45336f5a20ff87b68425b2e7fc181323dd/assets/src/edit-story/elements/image/karma/resourceLoading.karma.js#L43-L45

I have made the necessary changes, correct me if I am wrong. Thank you!

**Comment by Sreelayavuyyuru** [comment]
How do I solve this? Do I change `getByAriaLabel` to `screen.getByAltText` in the renderMediaElement too? 
Or have I done something wrong?

File: assets/src/edit-story/components/library/test/mediaElement.js [hunk 5]
```diff
@@ -18,6 +18,7 @@
  * External dependencies
  */
 import { Simulate } from 'react-dom/test-utils';
+import { render, screen } from '@testing-library/react';
```
Comment (swissspidy) [inline assets/src/edit-story/components/library/test/mediaElement.js]:
```suggestion
import { screen } from '@testing-library/react';
```

**Comment by swissspidy** [comment]
> How do I solve this? Do I change `getByAriaLabel` to `screen.getByAltText` in the renderMediaElement too?
> Or have I done something wrong?

Change `const { getByAriaLabel, queryByAriaLabel } = renderMediaElement(` to just `renderMediaElement(`. See the example I shared above:


```js
renderMediaElement(resource, 'local');
expect(screen.queryByAltText('More')).not.toBeInTheDocument();

const element = screen.getByAltText('image :)');
Simulate.focus(element);

expect(screen.getByAltText('More')).toBeInTheDocument();
```

See how there is no `getByAriaLabel` anywhere anymore

File: assets/src/edit-story/elements/image/karma/resourceLoading.karma.js [hunk 6]
```diff
@@ -40,7 +40,7 @@ describe('Image resource loading integration', () => {
     // Sleep a bit to ensure the media gallery grid is properly laid out.
     await fixture.events.sleep(50);
 
-    const image = fixture.screen.getAllByLabelText(
+    const image = fixture.screen.getByAltText() (
```
Comment (swissspidy) [inline assets/src/edit-story/elements/image/karma/resourceLoading.karma.js]:
```suggestion
    const image = fixture.screen.getByAltText(
```

File: assets/src/edit-story/components/library/test/mediaElement.js [hunk 7]
```diff
@@ -79,16 +80,13 @@ describe('MediaElement', () => {
       alt: 'image :)',
     };
 
-    const { getByAriaLabel, queryByAriaLabel } = renderMediaElement(
-      resource,
-      'local'
-    );
+    renderMediaElement(resource, 'local');
     expect(queryByAriaLabel('More')).not.toBeInTheDocument();
```
Comment (swissspidy) [inline assets/src/edit-story/components/library/test/mediaElement.js]:
```suggestion
    expect(screen.queryByAltText('More')).not.toBeInTheDocument();
```

File: assets/src/edit-story/components/library/test/mediaElement.js [hunk 8]
```diff
@@ -103,16 +101,13 @@ describe('MediaElement', () => {
       alt: 'video :)',
     };
 
-    const { getByAriaLabel, queryByAriaLabel } = renderMediaElement(
-      resource,
-      'local'
-    );
+    renderMediaElement(resource, 'local');
     expect(queryByAriaLabel('More')).not.toBeInTheDocument();
```
Comment (swissspidy) [inline assets/src/edit-story/components/library/test/mediaElement.js]:
```suggestion
    expect(screen.queryByAltText('More')).not.toBeInTheDocument();
```

File: assets/src/edit-story/components/library/test/mediaElement.js [hunk 9]
```diff
@@ -127,13 +122,10 @@ describe('MediaElement', () => {
       alt: 'image :)',
     };
 
-    const { getByAriaLabel, queryByAriaLabel } = renderMediaElement(
-      resource,
-      'local'
-    );
+    renderMediaElement(resource, 'local');
     expect(queryByAriaLabel('More')).not.toBeInTheDocument();
```
Comment (swissspidy) [inline assets/src/edit-story/components/library/test/mediaElement.js]:
```suggestion
    expect(screen.queryByAltText('More')).not.toBeInTheDocument();
```

File: assets/src/edit-story/components/library/test/mediaElement.js [hunk 10]
```diff
@@ -127,13 +122,10 @@ describe('MediaElement', () => {
...
 
-    const element = getByAriaLabel('image :)');
+    const element = screen.getByAltText('image :)');
     Simulate.focus(element);
 
     expect(queryByAriaLabel('More')).not.toBeInTheDocument();
```
Comment (swissspidy) [inline assets/src/edit-story/components/library/test/mediaElement.js]:
```suggestion
    expect(screen.queryByAltText('More')).not.toBeInTheDocument();
```

File: assets/src/edit-story/components/library/test/mediaElement.js [hunk 11]
```diff
@@ -151,13 +143,10 @@ describe('MediaElement', () => {
       alt: 'video :)',
     };
 
-    const { getByAriaLabel, queryByAriaLabel } = renderMediaElement(
-      resource,
-      'local'
-    );
+    renderMediaElement(resource, 'local');
     expect(queryByAriaLabel('More')).not.toBeInTheDocument();
```
Comment (swissspidy) [inline assets/src/edit-story/components/library/test/mediaElement.js]:
```suggestion
    expect(screen.queryByAltText('More')).not.toBeInTheDocument();
```

File: assets/src/edit-story/components/library/test/mediaElement.js [hunk 12]
```diff
@@ -151,13 +143,10 @@ describe('MediaElement', () => {
...
 
-    const element = getByAriaLabel('video :)');
+    const element = screen.getByAltText('video :)');
     Simulate.focus(element);
 
     expect(queryByAriaLabel('More')).not.toBeInTheDocument();
```
Comment (swissspidy) [inline assets/src/edit-story/components/library/test/mediaElement.js]:
```suggestion
    expect(screen.queryByAltText('More')).not.toBeInTheDocument();
```

**Comment by swissspidy** [comment]
I recommend running tests & ESLint locally first before pushing changes:

```
npx eslint assets/src/edit-story/components/library/test/mediaElement.js
npm run test:js assets/src/edit-story/components/library/test/mediaElement.js
```

**Comment by Sreelayavuyyuru** [comment]
I am not able to reproduce this issue, it would be better if anyone could take this up.

**Comment by swissspidy** [comment]
@Sreelayavuyyuru Sorry to see you closing this PR! :( I felt like it was really close to being done.

**Comment by Sreelayavuyyuru** [comment]
> @Sreelayavuyyuru Sorry to see you closing this PR! :( I felt like it was really close to being done.

I really felt that someone else could do it better, I'll try working on some other beginner-friendly issues :((

**Pull request closed without merging**